test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -535,13 +537,27 @@ protected void doAssertReducedMultiBucketConsumer(Aggregation agg, MultiBucketCo
     }
 
     protected <T extends AggregationBuilder, V extends InternalAggregation> void testCase(
-            T aggregationBuilder,
-            Query query,
-            CheckedConsumer<RandomIndexWriter, IOException> buildIndex,
-            Consumer<V> verify,
-            MappedFieldType... fieldTypes) throws IOException {
+        T aggregationBuilder,
+        Query query,
+        CheckedConsumer<RandomIndexWriter, IOException> buildIndex,
+        Consumer<V> verify,
+        MappedFieldType... fieldTypes) throws IOException {
+        testCase(aggregationBuilder, Optional.empty(), query, buildIndex, verify, fieldTypes);
+    }
+
+    protected <T extends AggregationBuilder, V extends InternalAggregation> void testCase(
+        T aggregationBuilder,
+        Optional<IndexWriterConfig> indexWriterConfig, Query query,
+        CheckedConsumer<RandomIndexWriter, IOException> buildIndex,
+        Consumer<V> verify,
+        MappedFieldType... fieldTypes) throws IOException {
         try (Directory directory = newDirectory()) {
-            RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
+            RandomIndexWriter indexWriter;
```

Comment:
I think Optional prefers things to look like:
```
RandomIndexWriter indexWriter = indexWriterConfg.map(c -> new RandomIndexWriter(random(), directory, c).orElseGet(() -> new RandomIndexWriter(random(), directory);
```

Reply:
Yeah, I agree that's what `Optional` would like this to look like.  Unfortunately, `new RandomIndexWriter` can throw an `IOException`, and the lambda `Optional#map` and friends expect is declared to not throw.  I could do something to wrap and throw an unchecked exception, but I think it'd be less clear than just using the if/else block.  Open to suggestions if you know a way to do this that isn't terrible.